lit_tests/compiler_plugin_basic.swift

```diff
@@ -0,0 +1,29 @@
+// REQUIRES: platform=Darwin
```

Comment:
Hmm, I was really hoping that we could get rid of the lit-based tests soon so that SwiftSyntax doesn’t have any dependency on tools in apple/swift. But it’s the best tool for the job here, so I think it’s fine for now.